### ISSUE TITLE
2D sweeps with the UHFQC_integration_logging_detector broken

### ISSUE THREAD

**Issue by stephlazar** [QudevETH/PycQED_py3#21]
At some point, we changed from setting MC.sweep points(nr_sweep_points x nr_shots) in the measure function and UHFQC.quex_rl_length(nr_shots) in the detector function, to setting MC.sweep points(nr_sweep_points) and UHFQC.quex_rl_length(nr_sweep_points x nr_shots).  The side effect of this is that now the MC.get_percdone() returns nr_shots % instead of 100 % when one acquisition is done.

This side effects broke the ability to do 2D sweeps with the SSRO detector because the MC only proceeds to the next acquisition in the 2D sweep is MC.get_percdone() < 100. 

To solve, we probably need to either:
- remove dependency of 2D sweep on this condition
- modify the get_percdone function to return correct percentage also for SSRO

**Comment by nathlacroix** [comment]
Thanks for raising the issue. My opinion is that we should address both points you have mentioned ;) 
- what was the purpose of the condition to continue only if MC.percdone() returned less than 100 ? Can we continue until we don't have any soft sweep points ? 

- ssro should return correct percentage also. Can we do that without an if statement ?

**Comment by antsr** [comment]
I merged the fix by Steph into qudev_master now, but it should still be tested that it is fixed on a real setup before closing

**Comment by antsr** [comment]
@stephlazar @nathlacroix Have you seen this issue this year?

**Comment by stephlazar** [comment]
yes I'm pretty sure I did. I didn't get to spend time to try to fix it though.
